compiler/src/main/java/com/squareup/anvil/compiler/codegen/dagger/DaggerGenerationUtils.kt

```diff
@@ -250,3 +265,86 @@ private fun KtTypeElement.singleTypeArgument(): KtTypeReference {
     .filterIsInstance<KtTypeReference>()
     .single()
 }
+
```

Comment:
These changes are almost all copy-pasta from MembersInjectionUtils.kt aside from some visibility changes and removal of redundant parameters for ClassReference extensions